### PR TITLE
Celery

### PR DESCRIPTION
**Pull request by halfak**
* Implements working score processors. Includes simple Timeout processor and Celery distributed processor.
* Adds stopit timer library to requirements.txt.
* Restructures the system to use 'ScoreProcessors' that contain 'ScoringContexts' for each wiki or group of wikis. 
* Updates ores-localdev.yaml to describe score_processors
* Adds celery_worker utility.
* Updates dev_server to work in threaded mode so that it can more easily take advantage of celery.

### CODE REVIEW

**Comment by halfak** [comment]
@yuvipanda refactoring and testing complete.  According to my tests, we'll have no problem keeping up with live Wikipedia now.  :)

File: ores/scoring_contexts/scoring_context.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+logger = logging.getLogger("ores.scorer.scorer")
+
+class ScoringContext(dict):
+
+    def __init__(self, name, scorer_models, extractor):
+        """
+        :Parameters:
+            name : str
+               The  that this scorer is being used for
```
Comment (he7d3r) [inline ores/scoring_contexts/scoring_context.py]: The that?

File: ores/scoring_contexts/scoring_context.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    def solve(self, model, cache):
+        """
+        Solves a model's features for a revision.  Does not attempt to gather
+        new data.
+        """
+        features = self[model].features
+        return self.extractor.solve(features, cache=cache)
+
+    def score(self, model, cache):
+        # TODO: record time spend computing features
```
Comment (he7d3r) [inline ores/scoring_contexts/scoring_context.py]: spent?

File: ores/scoring_contexts/scoring_context.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        start = time.time()
+        score = self[model].score(feature_values)
+        logger.debug("Scored features for {0}.{1} in {2} seconds"
+                     .format(self.name, model, time.time() - start))
+
+        return score
+
+    def extract_roots(self, model, rev_ids, caches=None):
+        """
+        Extracts a mapping of root datasources capabile of generating the
```
Comment (he7d3r) [inline ores/scoring_contexts/scoring_context.py]: capable?

File: ores/utilities/celery_worker.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
+"""
+Starts a celery worker for ORES.  Note that
```
Comment (he7d3r) [inline ores/utilities/celery_worker.py]: Not that what?

File: ores/wsgi/routes/scores.py [hunk 5]
```diff
@@ -7,32 +7,32 @@
...
-        wikis.sort()
+        contexts.sort()
 
-        return jsonify({'wikis': wikis})
+        return jsonify({'contexts': contexts})
 
     # /scores/enwiki/?models=reverted&revids=456789|4567890
-    @bp.route("/scores/<wiki>/", methods=["GET"])
-    def score_model_revisions(wiki):
+    @bp.route("/scores/<context>/", methods=["GET"])
```
Comment (he7d3r) [inline ores/wsgi/routes/scores.py]: Just for curiosity on the terminology: are there "contexts" which are not "wikis"? In other words, is a "context" a more general concept than a "wiki"?

**Pull request merged**